platform/admin/configurator/admin-configurator-impl/src/main/java/org/codice/ddf/admin/configurator/impl/BundleOperation.java

```diff
@@ -129,4 +152,30 @@ private BundleStateService getBundleStateService() {
     private boolean lookupBundleState() {
         return getBundleStateService().getState(bundle) == BundleState.Active;
     }
+
+    private boolean isPermittedToViewBundle(Bundle bundle) {
+        ServiceReference<?>[] services = bundle.getRegisteredServices();
+        if (services == null || services.length == 0) {
+            return true;
+        }
+
+        for (ServiceReference<?> service : services) {
+            Object pidProperty = service.getProperty("service.pid");
+            if (pidProperty instanceof String) {
+                if (!isPermittedToViewService((String) pidProperty)) {
```

Comment:
Could we just do `return isPermittedToViewService((String) pidProperty)` and remove the if statement?

Reply:
Could get a class cast in that case. Shouldn't. Ever. But could.

Reply:
If you keep the outer instanceof if statement there shouldn't be a class cast issue with removing the inner if

Reply:
Duh. Yeah. When I read your comment earlier, it was without the context of what's in the if block and I had forgotten it was exactly as dumb as it was. Of course I can change that.

Reply:
Nope. Scratch that. Looking at context I can't do that. I need to run that in a loop and I'm breaking out if and only if there is a service to which the Subject does not have permissions. If I change it to a return, I'll always return the response for the first service and never check the rest.